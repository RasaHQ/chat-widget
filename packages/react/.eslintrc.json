{
    "env": {
        "browser": true,
        "es2022": true,
        "es6": true
    },
    "extends": [
        "eslint:recommended",
        "plugin:@typescript-eslint/recommended",
        "plugin:react/recommended"
    ],
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "ecmaVersion": "latest",
        "sourceType": "module"
    },
    "ignorePatterns": [
        "dist",
        "lib/components/stencil-generated"
    ],
    "plugins": [
        "@typescript-eslint",
        "react"
    ],
    "rules": {}
}
