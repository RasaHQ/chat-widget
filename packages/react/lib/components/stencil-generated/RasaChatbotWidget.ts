'use client';

/**
 * This file was automatically generated by the Stencil React Output Target.
 * Changes to this file may cause incorrect behavior and will be lost if the code is regenerated.
 */

/* eslint-disable */

import type { EventName, StencilReactComponent } from '@stencil/react-output-target/runtime';
import { createComponent } from '@stencil/react-output-target/runtime';
import { RasaChatbotWidget as RasaChatbotWidgetElement, defineCustomElement as defineRasaChatbotWidget } from "@rasahq/chat-widget-ui/dist/components/rasa-chatbot-widget.js";
import React from 'react';

type RasaChatbotWidgetEvents = {
    onChatSessionStarted: EventName<CustomEvent<{ sessionId: string }>>,
    onChatWidgetReceivedMessage: EventName<CustomEvent<unknown>>,
    onChatWidgetSentMessage: EventName<CustomEvent<string>>,
    onChatWidgetQuickReply: EventName<CustomEvent<string>>,
    onChatWidgetOpened: EventName<CustomEvent<undefined>>,
    onChatWidgetClosed: EventName<CustomEvent<undefined>>,
    onChatWidgetHyperlinkClicked: EventName<CustomEvent<undefined>>,
    onChatWidgetFileStartedDownload: EventName<CustomEvent<undefined>>
};

const RasaChatbotWidget: StencilReactComponent<RasaChatbotWidgetElement, RasaChatbotWidgetEvents> = /*@__PURE__*/ createComponent<RasaChatbotWidgetElement, RasaChatbotWidgetEvents>({
    tagName: 'rasa-chatbot-widget',
    elementClass: RasaChatbotWidgetElement,
    react: React,
    events: {
        onChatSessionStarted: 'chatSessionStarted',
        onChatWidgetReceivedMessage: 'chatWidgetReceivedMessage',
        onChatWidgetSentMessage: 'chatWidgetSentMessage',
        onChatWidgetQuickReply: 'chatWidgetQuickReply',
        onChatWidgetOpened: 'chatWidgetOpened',
        onChatWidgetClosed: 'chatWidgetClosed',
        onChatWidgetHyperlinkClicked: 'chatWidgetHyperlinkClicked',
        onChatWidgetFileStartedDownload: 'chatWidgetFileStartedDownload'
    } as RasaChatbotWidgetEvents,
    defineCustomElement: defineRasaChatbotWidget
});

export default RasaChatbotWidget;
