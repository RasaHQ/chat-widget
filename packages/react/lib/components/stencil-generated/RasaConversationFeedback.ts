'use client';

/**
 * This file was automatically generated by the Stencil React Output Target.
 * Changes to this file may cause incorrect behavior and will be lost if the code is regenerated.
 */

/* eslint-disable */

import { RasaConversationFeedback as RasaConversationFeedbackElement, defineCustomElement as defineRasaConversationFeedback } from "@rasahq/chat-widget-ui/dist/components/rasa-conversation-feedback.js";
import type { EventName, StencilReactComponent } from '@stencil/react-output-target/runtime';
import { createComponent } from '@stencil/react-output-target/runtime';
import React from 'react';

type RasaConversationFeedbackEvents = { onFeedbackSubmitted: EventName<CustomEvent<{ rating: 'positive' | 'negative'; helpful: boolean }>> };

const RasaConversationFeedback: StencilReactComponent<RasaConversationFeedbackElement, RasaConversationFeedbackEvents> = /*@__PURE__*/ createComponent<RasaConversationFeedbackElement, RasaConversationFeedbackEvents>({
    tagName: 'rasa-conversation-feedback',
    elementClass: RasaConversationFeedbackElement,
    // @ts-ignore - React type of Stencil Output Target may differ from the React version used in the Nuxt.js project, this can be ignored.
    react: React,
    events: { onFeedbackSubmitted: 'feedbackSubmitted' } as RasaConversationFeedbackEvents,
    defineCustomElement: defineRasaConversationFeedback
});

export default RasaConversationFeedback;
