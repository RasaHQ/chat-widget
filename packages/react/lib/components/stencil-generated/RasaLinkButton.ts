'use client';

/**
 * This file was automatically generated by the Stencil React Output Target.
 * Changes to this file may cause incorrect behavior and will be lost if the code is regenerated.
 */

/* eslint-disable */

import { RasaLinkButton as RasaLinkButtonElement, defineCustomElement as defineRasaLinkButton } from "@rasahq/chat-widget-ui/dist/components/rasa-link-button.js";
import type { StencilReactComponent } from '@stencil/react-output-target/runtime';
import { createComponent } from '@stencil/react-output-target/runtime';
import React from 'react';

type RasaLinkButtonEvents = NonNullable<unknown>;

const RasaLinkButton: StencilReactComponent<RasaLinkButtonElement, RasaLinkButtonEvents> = /*@__PURE__*/ createComponent<RasaLinkButtonElement, RasaLinkButtonEvents>({
    tagName: 'rasa-link-button',
    elementClass: RasaLinkButtonElement,
    react: React,
    events: {} as RasaLinkButtonEvents,
    defineCustomElement: defineRasaLinkButton
});

export default RasaLinkButton;
