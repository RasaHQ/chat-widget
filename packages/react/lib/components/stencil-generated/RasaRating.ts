'use client';

/**
 * This file was automatically generated by the Stencil React Output Target.
 * Changes to this file may cause incorrect behavior and will be lost if the code is regenerated.
 */

/* eslint-disable */

import { RasaRating as RasaRatingElement, defineCustomElement as defineRasaRating } from "@rasahq/chat-widget-ui/dist/components/rasa-rating.js";
import type { EventName, StencilReactComponent } from '@stencil/react-output-target/runtime';
import { createComponent } from '@stencil/react-output-target/runtime';
import React from 'react';

type RasaRatingEvents = { onRatingSelected: EventName<CustomEvent<{ value: string }>> };

const RasaRating: StencilReactComponent<RasaRatingElement, RasaRatingEvents> = /*@__PURE__*/ createComponent<RasaRatingElement, RasaRatingEvents>({
    tagName: 'rasa-rating',
    elementClass: RasaRatingElement,
    react: React,
    events: { onRatingSelected: 'ratingSelected' } as RasaRatingEvents,
    defineCustomElement: defineRasaRating
});

export default RasaRating;
