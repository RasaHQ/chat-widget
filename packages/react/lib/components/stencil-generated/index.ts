/* eslint-disable */
/* tslint:disable */
/* auto-generated react proxies */
import { createReactComponent } from './react-component-lib';

import type { JSX } from 'web-component';



export const RasaChatInput = /*@__PURE__*/createReactComponent<JSX.RasaChatInput, HTMLRasaChatInputElement>('rasa-chat-input');
export const RasaChatbotWidget = /*@__PURE__*/createReactComponent<JSX.RasaChatbotWidget, HTMLRasaChatbotWidgetElement>('rasa-chatbot-widget');
export const RasaText = /*@__PURE__*/createReactComponent<JSX.RasaText, HTMLRasaTextElement>('rasa-text');
