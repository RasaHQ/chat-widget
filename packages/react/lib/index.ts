export * from "./components/stencil-generated";
export { defineCustomElements } from "@rasa-widget/web-component/loader";

import "@rasa-widget/web-component/dist/web-component/web-component.css";
