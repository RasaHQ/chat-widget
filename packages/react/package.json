{
  "name": "@vortexwest/chat-widget-react",
  "version": "0.1.15",
  "publishConfig": {
    "access": "restricted"
  },
  "description": "The Rasa Chatbot React Library enables seamless integration of Rasa-powered chatbot into React applications using web components.",
  "homepage": "https://github.com/RasaHQ/chat-widget#readme",
  "license": "ISC",
  "main": "dist/index.js",
  "module": "dist/index.js",
  "types": "dist/types/RasaChatbotWidget.d.ts",
  "directories": {
    "lib": "lib",
    "test": "__tests__"
  },
  "files": [
    "dist"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/RasaHQ/chat-widget.git"
  },
  "scripts": {
    "build": "npm run tsc",
    "tsc": "tsc -p . --outDir ./dist",
    "test": "jest"
  },
  "dependencies": {
    "@vortexwest/chat-widget-ui": "^0.1.15"
  },
  "devDependencies": {
    "@babel/preset-env": "^7.24.8",
    "@babel/preset-react": "^7.24.7",
    "@babel/preset-typescript": "^7.24.7",
    "@testing-library/dom": "^10.4.0",
    "@testing-library/jest-dom": "^6.4.7",
    "@testing-library/react": "^16.0.0",
    "@types/react": "^18.3.3",
    "@types/react-dom": "^18.3.0",
    "@typescript-eslint/eslint-plugin": "^7.17.0",
    "@typescript-eslint/parser": "^7.17.0",
    "babel-jest": "^29.7.0",
    "eslint": "^8.57.0",
    "eslint-plugin-react": "^7.34.1",
    "jest": "^29.7.0",
    "jest-environment-jsdom": "^29.7.0",
    "react": "^18.3.1",
    "react-dom": "^18.3.1",
    "typescript": "^5.5.4"
  }
}
