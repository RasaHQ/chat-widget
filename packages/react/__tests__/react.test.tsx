import React from "react";
import "@testing-library/jest-dom";
import { render, waitFor } from "@testing-library/react";

import { RasaChatbotWidget } from "../lib/components/stencil-generated";

describe("Rasa React Wrapper", () => {
  test("it renders web component", async () => {
    const { container } = render(
      <RasaChatbotWidget serverUrl="https://pro.example.com" />
    );

    const chatbotElement = container.querySelector("rasa-chatbot-widget");

    await waitFor(() => {
      expect(chatbotElement).toHaveClass("hydrated");
      expect(chatbotElement).toHaveAttribute(
        "server-url",
        "https://pro.example.com"
      );
      expect(chatbotElement).toMatchSnapshot();
    });
  });
});
