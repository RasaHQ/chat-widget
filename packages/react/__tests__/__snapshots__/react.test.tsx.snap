// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rasa React Wrapper it renders web component 1`] = `
<rasa-chatbot-widget
  class="hydrated"
  server-url="https://pro.example.com"
/>
`;
